src/state/slices/portfolioSlice/utils.ts

```diff
@@ -179,6 +185,26 @@ export const accountToPortfolio: AccountToPortfolio = args => {
           // nft assets will be dynamically upserted based on the state of the txHistory slice after the portfolio is loaded
           if (!isNft(token.assetId) && !args.assetIds.includes(token.assetId)) return
 
+          if (isNft(token.assetId)) {
+            const { assetReference, chainId, assetNamespace } = fromAssetId(token.assetId)
+            const [contractAddress] = deserializeNftAssetReference(assetReference)
+            const collectionAssetId = toAssetId({
+              chainId,
+              assetReference: contractAddress,
+              assetNamespace,
+            })
+
+            const nftCollection = nftCollectionsById[collectionAssetId]
+
+            if (nftCollection?.isSpam) return
+            if (
+              nftNameBlacklistRegex.test(
+                `${nftCollection?.description.toLowerCase()} ${nftCollection?.name.toLowerCase()}}`,
+              )
+            )
```

Comment:
something similar to this would be cool
https://github.com/shapeshift/web/pull/5309#discussion_r1329332977

Reply:
https://github.com/shapeshift/web/pull/5310/commits/e80a6fd7181c299433d4311aaf4218e4b4a04314